src/handler_forward.js

```diff
@@ -154,11 +154,43 @@ export default class HandlerForward extends HandlerBase {
 
         this.srcResponse.writeHead(response.statusCode, headers);
         response.pipe(this.srcResponse);
+
+        // Only detach on success, if there's an error
+        // it will be handled by `onTrgError` which calls `fail`,
+        // which forces the socket to disconnect.
+        response.once('end', () => {
+            this.detach();
+        });
     }
 
     onTrgError(err) {
         if (this.isClosed) return;
         this.log(`Target socket failed: ${err.stack || err}`);
         this.fail(err);
     }
+
+    /**
+     * Detach removes all listeners registered by HandlerBase.
+     * Must be called when the handler finishes on success,
+     * in order to prevent event emitter memory leak.
```

Comment:
Perhaps we could also add a link to https://github.com/apify/proxy-chain/pull/130

Reply:
done 👍🏼